test/jdk/javax/swing/JCheckBox/ImageCheckboxFocus/ImageCheckboxTest.java

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright (c) 2021, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import java.awt.BorderLayout;
+import java.awt.Color;
+import java.awt.Component;
+import java.awt.Graphics;
+import java.awt.Insets;
+import java.awt.Point;
+import java.awt.Rectangle;
+import java.awt.image.BufferedImage;
+import java.io.File;
+import javax.imageio.ImageIO;
+import javax.swing.BoxLayout;
+import javax.swing.Icon;
+import javax.swing.JButton;
+import javax.swing.JCheckBox;
+import javax.swing.JFrame;
+import javax.swing.JPanel;
+import javax.swing.SwingUtilities;
+import javax.swing.WindowConstants;
+
+import static java.awt.event.KeyEvent.VK_TAB;
+
+/*
+ * @test
+ * @key headful
+ * @bug 8216358
+ * @summary [macos] The focus is invisible when tab to "Image Radio Buttons" and "Image CheckBoxes"
+ * @library ../../regtesthelpers/
+ * @library /lib/client/
+ * @build Util
+ * @build ExtendedRobot
+ * @run main ImageCheckboxTest
+ */
+
+public class ImageCheckboxTest {
+    private static JFrame frame;
+    private static JButton testButton;
+    private static int locx, locy, frw, frh;
+
+    public static void main(String[] args) throws Exception {
+        new ImageCheckboxTest().performTest();
+    }
+
+    public void performTest() throws Exception {
+        try {
+            BufferedImage imageFocus1 = null;
```

Comment:
Probably the usage of Robot/Frame could be removed? Could you try to create the buffered image and button then paint the button to the graphics via button.paint(Graphics)? I guess in this case you can check the content of the buffered image directly w/o the robot.

Reply:
How it will work if the button was too small say w=4 pixels?

Reply:
1. Fixed.
2. If size of the icon based checkbox/radio button is less or equal than the icon size then the central part of the icon will be cropped and the halo will not be visible unless there are transparent pixels in the icon or icon is procedural and not paints on the whole icon area. We will not draw over other components in this case.